### PR TITLE
Unit tests & dependencies refactoring

### PR DESCRIPTION
**Pull request by marioradu05**

### CODE REVIEW

File: src/App/src/Middleware/AuthorizationMiddleware.php [hunk 1]
```diff
@@ -21,81 +21,65 @@
...
-                                sprintf(Message::USER_NOT_FOUND_BY_IDENTITY, $defaultUser->getIdentity())
-                            ]
-                        ]
-                    ], Response::STATUS_CODE_403);
+                    return $this->unauthorizedResponse(sprintf(
+                        Message::USER_NOT_FOUND_BY_IDENTITY,
+                        $defaultUser->getIdentity()
+                    ));
                 }
                 if ($user->getStatus() !== User::STATUS_ACTIVE) {
```
Comment (alexmerlin) [inline src/App/src/Middleware/AuthorizationMiddleware.php:81]: We could use replace this with:
```
if (!$user->isActive()) {
    return $this->unauthorizedResponse(Message::USER_NOT_ACTIVATED);
}
```

File: src/App/src/Middleware/AuthorizationMiddleware.php [hunk 2]
```diff
@@ -21,81 +21,65 @@
...
      */
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         /** @var UserIdentity $defaultUser */
         $defaultUser = $request->getAttribute(UserInterface::class);
         switch ($defaultUser->getDetail('oauth_client_id')) {
             case 'admin':
-                $user = $this->adminService->findByIdentity($defaultUser->getIdentity());
+                $user = $this->adminRepository->findOneBy(['identity' => $defaultUser->getIdentity()]);
                 if ($user->getStatus() !== Admin::STATUS_ACTIVE) {
```
Comment (alexmerlin) [inline src/App/src/Middleware/AuthorizationMiddleware.php]: Should implement the `isActive` method like in the `User` entity and use that to check if admin account is active:
```
if (!$user->isActive()) {
    return $this->unauthorizedResponse(Message::ADMIN_NOT_ACTIVATED);
}
```

File: tests/Unit/AdminServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Unit;
+
+use Api\Admin\Entity\Admin;
+use Api\Admin\Entity\AdminRole;
+use Api\Admin\Repository\AdminRepository;
+use Api\Admin\Service\AdminRoleService;
+use Api\Admin\Service\AdminService;
+use Api\Admin\Service\AdminService as Subject;
```
Comment (alexmerlin) [inline tests/Unit/AdminServiceTest.php:9]:
```
use Api\Admin\Service\AdminService;
use Api\Admin\Service\AdminService as Subject;
```
Do you need both of them?

**Review by alexmerlin — COMMENTED** [review]
Also, please remove the extra empty line at the beginning of the following files:
* tests/Unit/AuthorizationMiddlewareTest.php
* tests/Unit/AuthenticationMiddlewareTest.php
* src/User/src/Service/UserAvatarService.php

File: tests/Unit/AdminServiceTest.php (hunk 3 above)
Comment (marioradu05) [reply tests/Unit/AdminServiceTest.php:9]: no, i ve removed `use Api\Admin\Service\AdminService;
`

**Review by alexmerlin — APPROVED** [review]

**Pull request merged**